### DISCUSSION TITLE
Slider for input values

### DISCUSSION THREAD

**Discussion by Hietakissa** [kazad/instacalc#75] [Ideas]
Having the option to give a variable a value using a slider would be very nice for inputs. 
The format could look something like: "num = slider(min,max)", where the num variable would have a slider next to it you can use to set a value between the given min and max values. It could also be a formatting option for the row.

**Comment by kazad** [comment]
Great idea, thanks. I've long wanted to have sliders, checkboxes, radio buttons, dropdowns, etc. I'd like to do some research and see if there's a clean, well-designed API for that sort of thing.

`slider(min, max, default)`
`slider(min:1, max:10, default: 3)`

Part of this may entail having objects and arrays as first-class items in instacalc. Appreciate the suggestion.

**Comment by kazad** [comment]
Making progress on this one, I have some internal prototypes. 

The main syntax is

`x = 10 // {ui: ... }`

so you set a default value (like any variable), but in the comment you can specify what type of UI you want for that row. Currently playing with the regular HTML native elements (sliders, checkboxes, dropdowns, radios, date pickers, etc.)

<img width="2062" height="1124" alt="image" src="https://github.com/user-attachments/assets/9bd22425-de95-484e-910a-a9c2754078bd" />

**Reply by kazad** [reply]
Other notes/goals

* backwards compatible with regular text, the UI is just an enhancement. You can still modify row text directly when in other modes (like text mode)
* use native HTML controls where possible (but improved styles?)

**Comment by kazad** [comment]
Shipped in beta, see https://github.com/kazad/instacalc/discussions/137
